octoprint_mrbeam/static/js/laserhead_changed.js

```diff
@@ -0,0 +1,60 @@
+$(function () {
+    function LaserheadChangedViewModel(params) {
+        let self = this;
+        window.mrbeam.viewModels["LaserheadChangedViewModel"] = self;
+
+        self.settings = params[0];
+        self.loginState = params[1];
+
+        self.onUserLoggedIn = function () {
+            if (
+                self.loginState.currentUser?.()?.active
+            ) {
+                let laserheadChanged = self.settings.settings.plugins.mrbeam.laserheadChanged();
+                if (laserheadChanged) {
+                    $("#laserhead_changed").modal("show");
+                }
+            }
+        };
+
+        self.laserheadChangedDetected = function (){
+            let data = {
+                laserheadChanged: false,
+            };
+            OctoPrint.simpleApiCommand("mrbeam", "laserhead_changed", data)
+                .done(function (response) {
+                    self.settings.requestData();
+                    console.log(
+                        "simpleApiCall response for saving laserhead change detection: ",
+                        response
+                    );
+                })
+                .fail(function () {
+                    self.settings.requestData();
+                    console.error("Unable to save laserhead change detection state: ", data);
+                    new PNotify({
+                        title: gettext("Error while saving laserhead change detection!"),
+                        text: _.sprintf(
+                            gettext(
+                                "Unable to save laserhead change detection at the moment.%(br)sCheck connection to Mr Beam and try again."
+                            ),
+                            { br: "<br/>" }
+                        ),
+                        type: "error",
+                        hide: true,
+                    });
+                });
+        }
+    }
```

Comment:
What is the purpose of this? Why is the front end telling the back-end whether the laserhead got changed?

Reply:
If the data is static... Couldn't we ignore it all together?

Reply:
I think I understand what this is about now... Could you rename it to `laserheadChangeAcknowledged` ?

Reply:
I was think about Acknowledged as well ^^ 
Done

Reply:
Could you also reflect the new name in the json you send please?

Reply:
Or do not send any "data", just send back empty data with the API call `laserhead_change_acknowledged`

Reply:
No I think that stands correct. The config parameter reflect an event happening which is laser being changed

Reply:
And what is that for?
`self.settings.requestData();`

Reply:
Yes, in the config itself, not in the front-end. The front end only knows that the user acknowledged the change, but it is up to the backend to choose whether to do something with the acknowledgement

Reply:
self.settings.requestData() has been removed

Reply:
I agree with you, I just don't see what the need to change the parameter name into acknowledged in the backend